### PR TITLE
[CALCITE-6698] Add Javadoc to PartiallyOrderedSet#getNonChildren and getNonParents to clarify their behavior

### PR DESCRIPTION
**Pull request by zabetak**
1. Add Javadoc to clarify the results of each method and avoid potential misinterpretation of their name.
2. Add some test cases with explicit tests for the aforementioned methods.

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/util/PartiallyOrderedSetTest.java [hunk 1]
```diff
@@ -180,6 +181,32 @@ private static boolean isBitSuperset(Integer e1, Integer e2) {
...
+    PartiallyOrderedSet<Integer> poset =
+        new PartiallyOrderedSet<>((i, j) -> i >= j, Arrays.asList(20, 30, 40));
+    assertThat(poset.getNonParents(), hasToString("[20]"));
+  }
+
+  @Test void testGetNonChildrenOnGteIntPosetReturnsMaxValue() {
+    PartiallyOrderedSet<Integer> poset =
+        new PartiallyOrderedSet<>((i, j) -> i >= j, Arrays.asList(20, 30, 40));
+    printValidate(poset);
+    System.out.println(poset.getChildren(20));
```
Comment (asolimando) [inline core/src/test/java/org/apache/calcite/util/PartiallyOrderedSetTest.java:206]: This has slipped through the cracks I guess

**Comment by julianhyde** [comment]
-1 There is no bug here. You are just mistaken what 'non-parents' means.

File: core/src/test/java/org/apache/calcite/util/PartiallyOrderedSetTest.java (hunk 1 above)
Comment (zabetak) [reply core/src/test/java/org/apache/calcite/util/PartiallyOrderedSetTest.java]: Thanks for catching this. I removed those in the last commit.

**Comment by zabetak** [comment]
As discussed, under the ticket there is no reason to change the behavior of the methods. The scope of the PR changed to simply clarify the current behavior by adding Javadoc comments so I updated the summary and description.

**Pull request closed without merging**